server/src/ai/ai.router.ts

```diff
@@ -1,48 +1,116 @@
 import express from 'express';
 import { CourseIdRequestType } from '../course/course.type';
-import {
-  getCoursesWithMinReviews,
-  getReviewsForSummary,
-  generateTags
-} from './ai.functions';
+import { getCoursesWithMinReviews, getReviewsForSummary, summarize, updateCourseWithAI, getCoursesWithMinFreshness } from './ai.functions';
 
 const aiRouter = express.Router();
 
 aiRouter.use(express.json());
 
-/** Reachable at POST /api/ai/summarize-reviews
+/** Reachable at POST /api/ai/summarize-courses
+ * retrieves all course IDs with a minimum of 3 reviews,
+ * and then attempts to update the classSummary, classTags, and freshness
+ * for each course ID by calling updateCoursesWithAI. Makes at most 3 attempts
+ * to summarize course IDs with incomplete summary or tags. 
+ * returns a message that update has been completed and returns results which
+ * shows the courses that were summarized completely and the courses that were
+ * not.
+ */
+aiRouter.post('/summarize-courses', async (req, res) => {
+  try {
+    const minReviews = 3;
+    //get all courses with at least minimum reviews (will be changed later to check for freshness)
+    // const minFreshness = 3
+    // const courseIds = await getCoursesWithMinFreshness(minFreshness);
+    const courseIds = await getCoursesWithMinReviews(minReviews);
+    const results = { success: [], incomplete: [...courseIds] };
+    if (!courseIds || courseIds.length === 0) {
+      return res.json({ message: `No courses found with at least ${minReviews} reviews.`, results });
+    }
+    let currentIteration = 0;
+    let counter = 0;
+    while (results.incomplete.length > 0 && currentIteration < 3) {
+      const nextIncomplete = [];
+
+      for (const courseId of results.incomplete) {
+        const success = await updateCourseWithAI(courseId);
+        if (success) {
+          results.success.push(courseId);
+        } else {
+          nextIncomplete.push(courseId);
+        }
+        counter += 1;
+        if (counter == 5) {
```

Comment:
Is this to facilitate the case where a course always falls under the incomplete case?